core/src/test/java/com/rudderstack/android/sdk/core/RudderGsonTest.java

```diff
@@ -0,0 +1,271 @@
+
+@RunWith(PowerMockRunner.class)
+@PowerMockIgnore("jdk.internal.reflect.*")
+@PrepareForTest({Utils.class})
+public class RudderGsonTest {
+
+    @Before
+    public void setUp() throws Exception {
+        PowerMockito.spy(Utils.class);
+        PowerMockito.when(Utils.class, "getTimeStamp")
+                .thenAnswer((Answer<String>) invocation -> "2022-03-14T06:46:41.365Z");
+    }
+
+    @Test
+    public void testRudderContextSerializationSynchronicity() {
+        AtomicInteger contextsSerialized = new AtomicInteger(0);
+        new Thread() {
+            @Override
+            public void run() {
+                super.run();
+                for (int i = 1; i <= 1000; i++) {
+                    RudderContext context = getRudderContext(i);
+                    assertThat("contexts got serialized perfectly " + i, areJsonStringsEqual(RudderGson.getInstance().toJson(context), getRudderContextJsonString(i)), is(true));
+                    contextsSerialized.addAndGet(1);
+                }
+            }
+        }.start();
+        new Thread() {
+            @Override
+            public void run() {
+                super.run();
+                for (int i = 1001; i <= 2000; i++) {
+                    RudderContext context = getRudderContext(i);
+                    assertThat("contexts got serialized perfectly", areJsonStringsEqual(RudderGson.getInstance().toJson(context), getRudderContextJsonString(i)), is(true));
+                    contextsSerialized.addAndGet(1);
+                }
+            }
+        }.start();
+        new Thread() {
+            @Override
+            public void run() {
+                super.run();
+                for (int i = 2001; i <= 3000; i++) {
+                    RudderContext context = getRudderContext(i);
+                    assertThat("contexts got serialized perfectly", areJsonStringsEqual(RudderGson.getInstance().toJson(context), getRudderContextJsonString(i)), is(true));
+                    contextsSerialized.addAndGet(1);
+                }
+            }
+        }.start();
+        await().atMost(60, SECONDS).until(() -> contextsSerialized.get() == 3000);
+    }
+
+    private RudderContext getRudderContext(int i) {
+        RudderContext context = getDefaultRudderContext();
+        Map<String, Object> customContextMap = new HashMap<>();
+        Map<String, String> companyContext = new HashMap<>();
+        companyContext.put("name", "company" + i);
+        companyContext.put("city", "city" + i);
+        customContextMap.put("company", companyContext);
+        Map<String, String> personContext = new HashMap<>();
+        personContext.put("tier", "enterprise" + i);
+        customContextMap.put("person", personContext);
+        Whitebox.setInternalState(context, "customContextMap", customContextMap);
+        return context;
+    }
+
+    private RudderContext getDefaultRudderContext() {
+        RudderContext context = PowerMockito.mock(RudderContext.class);
```

Comment:
Is it possible to replace `PowerMockito` with `Mockito` here?  Can we stop adding `PowerMockito` to our tests? We won't be able to upgrade to an upper Java version otherwise.

Reply:
Sure, will check it and revert back

Reply:
replaced usage of powermockito with mockito, please take a look again